### ISSUE TITLE
Whisper no_speech_threshold not applied when chunking input

### ISSUE THREAD

**Issue by stri8ed** [huggingface/transformers#29595]
### Feature request

The Whisper pipeline accepts a `chunk_length_s` parameter, which chunks the input so it can be used for batch inference. There is also a `no_speech_threshold` param, which can be used to filter out silence, which helps with reducing halucinations. The problem is, for `no_speech_threshold` to be applied the [output must be "long"](https://github.com/huggingface/transformers/blob/e5eb55b88b7200e81144eaf7a2bf3001b2bd08e7/src/transformers/models/whisper/generation_whisper.py#L492). And in the case of chunked input, every segment is considered short, even though the full input is long. This means the `no_speech_threshold` cant be applied for chunked input.

When attempting, It gives this error for each batch:
`Audio input consists of only 3000. Short-form transcription is activated.no_speech_threshold is set to 0.2, but will be ignored.`

It should be possible to keep the no speech threshold enabled for long chunked inputs.

### Motivation

Chunking brings large performance improvements for long-form transcription, but this benefit is negated if there is no way to suppress the silence segments, which most long-form audio will no doubt contain.

### Your contribution

I could implement a change which removes the `is_shortform` check when the input is chunked, though I'm unsure if and why this will break things.

**Comment by amyeroberts** [comment]
cc @sanchit-gandhi @ylacombe

**Comment by stri8ed** [comment]
I was able to get around this, by using the Vad implementation from faster-whisper.

```
audio_data = load_audio(audio_file)
speech_chunks = get_speech_timestamps(audio_data, VadOptions())
audio_without_silence = collect_chunks(audio_data, speech_chunks)

prediction = pipe(audio_without_silence , ...)

chunks = prediction["chunks"]
ts_map = SpeechTimestampsMap(speech_chunks, sampling_rate)
segments = []
    for chunk in chunks:
        start, end = chunk["timestamp"]
        segments.append({
            "start": ts_map.get_original_time(start),
            "end": ts_map.get_original_time(end),
            "text": chunk["text"]
        })
```

**Comment by Kimahriman** [comment]
Related to https://github.com/huggingface/transformers/issues/29508, not sure why there are different implementations for long and short audio. I'm playing around with combining them

**Comment by amyeroberts** [comment]
Gentle ping @sanchit-gandhi @ylacombe

**Comment by amyeroberts** [comment]
Another ping @ylacombe @sanchit-gandhi @kashif

**Comment by sanchit-gandhi** [comment]
Indeed, as @Kimahriman mentioned in #29508 there should be no distinction between the short and long-form algorithms. This issue will be fixed when #29508 is fixed, by merging the short/long-form generation logic together.

**Comment by amyeroberts** [comment]
Is this fixed @kamilakesbi @sanchit-gandhi now #29508 is merged?

**Comment by amyeroberts** [comment]
cc @ylacombe

**Comment by ylacombe** [comment]
Hey @amyeroberts, this is indeed fixed since #29508.

@stri8ed, note that you also have to specify the float `logprob_threshold` and the list `temperature`, as indicated in [the docs](https://huggingface.co/docs/transformers/model_doc/whisper#transformers.WhisperForConditionalGeneration.generate) !
